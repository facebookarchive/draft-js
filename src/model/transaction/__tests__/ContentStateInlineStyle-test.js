/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall draft_js
 */

'use strict';
import type ContentState from 'ContentState';
import type SelectionState from 'SelectionState';

const ContentStateInlineStyle = require('ContentStateInlineStyle');

const getSampleStateForTesting = require('getSampleStateForTesting');

const {contentState, selectionState} = getSampleStateForTesting();

const initialSelection = selectionState.set(
  'focusOffset',
  contentState.getBlockForKey(selectionState.getStartKey()).getLength(),
);

const assertAddContentStateInlineStyle = (
  inlineStyle: $TEMPORARY$string<'BOLD'> | $TEMPORARY$string<'ITALIC'>,
  selection: $FlowFixMe | SelectionState = selectionState,
  content: ContentState = contentState,
) => {
  const newContentState = ContentStateInlineStyle.add(
    content,
    selection,
    inlineStyle,
  );

  expect(newContentState.getBlockMap().toJS()).toMatchSnapshot();

  return newContentState;
};

const assertRemoveContentStateInlineStyle = (
  inlineStyle: $TEMPORARY$string<'BOLD'> | $TEMPORARY$string<'ITALIC'>,
  selection: $FlowFixMe | SelectionState = selectionState,
  content: ContentState = contentState,
) => {
  const newContentState = ContentStateInlineStyle.remove(
    content,
    selection,
    inlineStyle,
  );

  expect(newContentState.getBlockMap().toJS()).toMatchSnapshot();

  return newContentState;
};

test('must add styles', () => {
  const modified = assertAddContentStateInlineStyle('BOLD', initialSelection);

  assertAddContentStateInlineStyle(
    'ITALIC',
    selectionState.set('focusOffset', 2),
    modified,
  );
});

test('must remove styles', () => {
  // Go ahead and add some styles that we'll then remove.
  let modified = assertAddContentStateInlineStyle('BOLD', initialSelection);
  modified = assertAddContentStateInlineStyle(
    'ITALIC',
    initialSelection,
    modified,
  );

  // we then remove the added styles
  modified = assertRemoveContentStateInlineStyle(
    'BOLD',
    initialSelection,
    modified,
  );
  assertRemoveContentStateInlineStyle(
    'ITALIC',
    initialSelection.set('focusOffset', 2),
    modified,
  );
});

test('must add and remove styles accross multiple blocks', () => {
  const nextBlock = contentState.getBlockAfter(selectionState.getStartKey());
  const selection = selectionState.merge({
    focusKey: nextBlock?.getKey(),
    focusOffset: nextBlock?.getLength(),
  });

  const modified = assertAddContentStateInlineStyle('BOLD', selection);
  assertRemoveContentStateInlineStyle('BOLD', selection, modified);
});
