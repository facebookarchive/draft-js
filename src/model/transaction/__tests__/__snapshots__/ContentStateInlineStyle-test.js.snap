// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must add and remove styles accross multiple blocks 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must add and remove styles accross multiple blocks 2`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [],
      },
      Object {
        "entity": "2",
        "style": Array [],
      },
      Object {
        "entity": "2",
        "style": Array [],
      },
      Object {
        "entity": "2",
        "style": Array [],
      },
      Object {
        "entity": "2",
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must add styles 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must add styles 2`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must remove styles 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must remove styles 2`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "BOLD",
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must remove styles 3`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must remove styles 4`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Test",
    "type": "code-block",
  },
  "d": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "d",
    "text": "",
    "type": "code-block",
  },
  "e": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "e",
    "text": "",
    "type": "code-block",
  },
  "f": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "f",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;
